### PR TITLE
Comment documentation

### PR DESCRIPTION
**Pull request by dilsgeorge**
Kindly review the part c of the project:

files changed:-

dils:
receiver_cadmium.hpp
test/src/receiver/main.cpp

Namitha:
test/src/sender/main.cpp
test/src/subnet/main.cpp

### CODE REVIEW

**Review by SREEJITH1991 — COMMENTED** [review]
delete README.txt since you already have README.md

**Comment by Namithajacob** [comment]
README.txt file is deleted based on the review comment.

**Review by SREEJITH1991 — APPROVED** [review]

File: include/receiver_cadmium.hpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	/**< putting definitions in context*/
+    using defs=receiver_defs;
+    public:
+    /**<This constant has the value of time delay between input and output*/
+    TIME   PREPARATION_TIME;
+    /**
+     * Constructor for receiver
+     * It initialize time delay constant with a value
+     */
+    Receiver() noexcept{
```
Comment (mitalraithatha) [inline include/receiver_cadmium.hpp:68]: This constructor also initialize ack_number and sending state.
So please add that also in comment.

File: include/receiver_cadmium.hpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        state.sending = false;
+    }
+
+    /**
+     * This function receives the message and checks the number of
+     * message. If the number of message is greater than 1, it says
+     * only one message per time unit. Else, it will set the receiver
+     * sending state as on, i.e active
+     * @param variable of type TIME
+     * @param varable of type make_message_bags
```
Comment (mitalraithatha) [inline include/receiver_cadmium.hpp]: Variable spelling is not correct. please correct that.

File: include/receiver_cadmium.hpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    void internal_transition() {
+        state.sending = false;
+    }
+
+    /**
+     * This function receives the message and checks the number of
+     * message. If the number of message is greater than 1, it says
+     * only one message per time unit. Else, it will set the receiver
+     * sending state as on, i.e active
+     * @param variable of type TIME
```
Comment (mitalraithatha) [inline include/receiver_cadmium.hpp]: Need to mention variable name (i.e : e) after @param .

File: include/receiver_cadmium.hpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                state.sending = true;
+            }
+                           
+    }
+
+    /**
+     * This function calls both internal_transition and external_transition
+     * functions.
+     * @param variable of type TIME
+     * @param varable of type make_message_bags
```
Comment (mitalraithatha) [inline include/receiver_cadmium.hpp]: Variable spelling is not correct. please correct that.

File: include/receiver_cadmium.hpp [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                state.ack_num = static_cast < int > (x.value);
+                state.sending = true;
+            }
+                           
+    }
+
+    /**
+     * This function calls both internal_transition and external_transition
+     * functions.
+     * @param variable of type TIME
```
Comment (mitalraithatha) [inline include/receiver_cadmium.hpp]: Need to mention variable name (i.e : e) after @param .

File: include/receiver_cadmium.hpp [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    	typename make_message_bags<input_ports>::type mbs) {
+        internal_transition();
+        external_transition(TIME(), std::move(mbs));
+    }
+
+    /**
+     * This function sends acknowledgment to the output port
+     * The acknowledgment value is calculated by the modulo of
+     * acknowledgment number with 10.
+     * @return variable of type make_message_bags
```
Comment (mitalraithatha) [inline include/receiver_cadmium.hpp]: Need to mention variable name (i.e : bags) after @return.

File: test/src/receiver/main.cpp [hunk 7]
```diff
@@ -1,3 +1,27 @@
+/**
+ * \brief  This main file of receiver implements the
+ * operation of receiver model which can be used for testing.
+ *
+ * receiver receive the data and send back an acknowledgement extracted
```
Comment (mitalraithatha) [inline test/src/receiver/main.cpp]: Spelling mistake in acknowledgment. Please correct that.

File: test/src/receiver/main.cpp [hunk 8]
```diff
@@ -16,45 +40,82 @@
...
+
+	/**
+     * Default constructor for the class
+	 */
+
     ApplicationGen() = default;
-    ApplicationGen(const char* file_path) : iestream_input<Message_t,
+
+    /**
+     * The below constructor of ApplicationGen class takes the input file path for
```
Comment (mitalraithatha) [inline test/src/receiver/main.cpp]: It is a "parameterized constructor".
add this name of constructor.

File: test/src/receiver/main.cpp [hunk 9]
```diff
@@ -16,45 +40,82 @@
...
         T>(file_path) {}
-    };
+};
 
 
 int main(){
 
-    auto start = hclock::now(); //to measure simulation execution time
+	/**
+	 *  measures simulation execution time
```
Comment (mitalraithatha) [inline test/src/receiver/main.cpp]: It is not measuring simulation execution time but initialize start variable to current time.

File: test/src/receiver/main.cpp [hunk 10]
```diff
@@ -131,19 +199,31 @@ int main(){
         ics_TOP
     );
 
-///****************////
+    /**
+     * Creates a model and measures the time taken to create the model created.
```
Comment (mitalraithatha) [inline test/src/receiver/main.cpp]: not clear comment. Please rephrase it.

File: test/src/receiver/main.cpp [hunk 11]
```diff
@@ -131,19 +199,31 @@ int main(){
...
+     */
 
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double,
+    auto time_elapsed = std::chrono::duration_cast<std::chrono::duration<double,
     		        std::ratio<1>>>(hclock::now() - start).count();
-    cout << "Model Created. Elapsed time: " << elapsed1 << "sec" << endl;
+    cout << "Model Created. Elapsed time: " << time_elapsed << "sec" << endl;
+
+    /**
+     *  This creates a runner and measures the time taken to create the same.
```
Comment (mitalraithatha) [inline test/src/receiver/main.cpp]: not clear comment. Please rephrase it

File: test/src/sender/main.cpp [hunk 12]
```diff
@@ -18,49 +51,100 @@
...
+
+	/**
+	 * Default constructor for the class
+	 */
+
     ApplicationGen() = default;
-    ApplicationGen(const char* file_path) : iestream_input<Message_t,
+
+    /**
+     * The below constructor of ApplicationGen class takes the input file path for
```
Comment (mitalraithatha) [inline test/src/sender/main.cpp]: It is a "parameterized constructor".
add this name of constructor.

File: test/src/sender/main.cpp [hunk 13]
```diff
@@ -156,22 +254,32 @@ int main(){
         ics_TOP
     );
 
-///****************////
+    /**
+     * Creates a model and measures the time taken to create the model created.
```
Comment (mitalraithatha) [inline test/src/sender/main.cpp]: not clear comment. Please rephrase it.

File: test/src/sender/main.cpp [hunk 14]
```diff
@@ -156,22 +254,32 @@ int main(){
...
+     */
 
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double,
+    auto time_elapsed = std::chrono::duration_cast<std::chrono::duration<double,
     	            std::ratio<1>>>(hclock::now() - start).count();
-    cout << "Model Created. Elapsed time: " << elapsed1 << "sec" << endl;
+    cout << "Model Created. Elapsed time: " << time_elapsed << "sec" << endl;
+
+    /**
+     *  This creates a runner and measures the time taken to create the same.
```
Comment (mitalraithatha) [inline test/src/sender/main.cpp]: not clear comment. Please rephrase it.

File: test/src/subnet/main.cpp [hunk 15]
```diff
@@ -134,22 +205,32 @@ int main(){
         ics_TOP
     );
 
-///****************////
+    /**
+     * Creates a model and measures the time taken to create the model created.
```
Comment (mitalraithatha) [inline test/src/subnet/main.cpp]: not clear comment. Please rephrase it.

File: test/src/subnet/main.cpp [hunk 16]
```diff
@@ -134,22 +205,32 @@ int main(){
...
+     */
 
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double,
+    auto time_elapsed = std::chrono::duration_cast<std::chrono::duration<double,
                     std::ratio<1>>>(hclock::now() - start).count();
-    cout << "Model Created. Elapsed time: " << elapsed1 << "sec" << endl;
+    cout << "Model Created. Elapsed time: " << time_elapsed << "sec" << endl;
     
+    /*
+     *  This creates a runner and measures the time taken to create the same.
```
Comment (mitalraithatha) [inline test/src/subnet/main.cpp]: not clear comment. Please rephrase it.

File: test/src/subnet/main.cpp [hunk 17]
```diff
@@ -18,45 +43,84 @@
...
+
+	/**
+	 * Default constructor for class
+	 */
+
     ApplicationGen() = default;
-    ApplicationGen(const char* file_path) : iestream_input<Message_t,
+
+    /**
+     * The below constructor of ApplicationGen class takes the input file path for
```
Comment (mitalraithatha) [inline test/src/subnet/main.cpp]: It is a "parameterized constructor".
add this name of constructor.

File: include/receiver_cadmium.hpp (hunk 1 above)
Comment (Namithajacob) [reply include/receiver_cadmium.hpp:68]: Comment was modified as suggested

File: include/receiver_cadmium.hpp (hunk 2 above)
Comment (Namithajacob) [reply include/receiver_cadmium.hpp]: spelling mistake is resolved

File: include/receiver_cadmium.hpp (hunk 3 above)
Comment (Namithajacob) [reply include/receiver_cadmium.hpp]: Mentioned variable name as suggested

File: include/receiver_cadmium.hpp (hunk 4 above)
Comment (Namithajacob) [reply include/receiver_cadmium.hpp]: Spelling mistake is resolved

File: include/receiver_cadmium.hpp (hunk 5 above)
Comment (Namithajacob) [reply include/receiver_cadmium.hpp]: mentioned variable name as suggested

File: test/src/sender/main.cpp (hunk 13 above)
Comment (Namithajacob) [reply test/src/sender/main.cpp]: Rephrased the comment as suggested

File: test/src/sender/main.cpp (hunk 14 above)
Comment (Namithajacob) [reply test/src/sender/main.cpp]: Rephrased the comment as suggested

File: test/src/subnet/main.cpp (hunk 15 above)
Comment (Namithajacob) [reply test/src/subnet/main.cpp]: Rephrased the comment as suggested

File: test/src/subnet/main.cpp (hunk 16 above)
Comment (Namithajacob) [reply test/src/subnet/main.cpp]: Rephrased the comment as suggested

File: test/src/subnet/main.cpp (hunk 17 above)
Comment (Namithajacob) [reply test/src/subnet/main.cpp]: Added the constructor name

**Review by mitalraithatha — APPROVED** [review]
Changes done according to review.
So Approving it.

**Pull request merged**